subprojects/logging/src/main/java/org/gradle/internal/featurelifecycle/LoggingDeprecatedFeatureHandler.java

```diff
@@ -76,9 +80,9 @@ public void deprecatedFeatureUsed(DeprecatedFeatureUsage usage) {
 
     public void reportSuppressedDeprecations() {
         if (warningMode == WarningMode.Summary && !messages.isEmpty()) {
-            LOGGER.warn("\nThere are {} deprecation warnings, which may break the build in Gradle {}. Please run with --warning-mode=all to see them.",
-                messages.size(),
-                GradleVersion.current().getNextMajor().getVersion());
+            LOGGER.warn("\n{} {}.\n{} {}",
+                WARNING_SUMMARY, GradleVersion.current().getNextMajor().getVersion(),
+                WARNING_LOGGING_DOCS_MESSAGE, DOCUMENTATION_REGISTRY.getDocumentationFor("command_line_interface", "sec:command_line_warnings"));
```

Comment:
The reference to documentation seems somewhat brittle. I doubt that we'll remember to change this code in case we'd ever rename chapter in the docs.

Reply:
This is something I'd like to do more of. We should maybe double-check all usages of DocumentationRegistry to avoid broken links (or add testing)

Reply:
Maybe at build time we could create an index of all identifies available in the user manual before compiling the classes and compare it to the usage.

Reply:
That idea sounds like it's worth spiking.